src/collections/components/StoryForm/StoryForm.tsx

```diff
@@ -82,6 +102,28 @@ export const StoryForm: React.FC<StoryFormProps & SharedFormButtonsProps> = (
   const [showOtherFields, setShowOtherFields] =
     useState<boolean>(showAllFields);
 
+  // Whether to show the full form or just the URL field with the "Populate" button
+  const [ogExcerptText, setOgExcerptText] = useState<string>('');
+  const [parserItemExcerptText, setParserItemExcerptText] =
+    useState<string>('');
+  const [descriptionState, setDescriptionState] =
+    useState<DescriptionTextStates>(
+      DescriptionTextStates.IncompleteDescription
+    );
+
+  useEffect(() => {
+    console.log(`OG: ${ogExcerptText} Parser: ${parserItemExcerptText}`);
+    if (ogExcerptText && parserItemExcerptText) {
+      if (ogExcerptText === parserItemExcerptText) {
+        setDescriptionState(DescriptionTextStates.MatchingDescriptions);
+      } else {
+        setDescriptionState(DescriptionTextStates.CanInsertParserDesc);
+      }
+    } else {
+      setDescriptionState(DescriptionTextStates.IncompleteDescription);
+    }
+  }, [ogExcerptText, parserItemExcerptText]);
```

Comment:
I am wondering if we really need a `useEffect()` here? We try to limit the use of this hook and try to use it where we absolutely need it. Do you think we just have this logic (line 116 - 124) in a just a regular function, given we will always have access to the state variables in this hook's dependency array? 

Just a thought!

Reply:
Description state can be changed from user interaction OR from new data coming in, so it is really needs this treatment. The code could be simpler if I could get the formik text field data, though I think the users might not like some of the behavior quirks. Let me know if you want to discuss further

Reply:
Sounds good and I think it does make sense here given that new data could come in.